From 2e65084c0bdcb88f670ce8ca2609802036809566 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Wed, 31 Jan 2024 22:48:22 +0100
Subject: [PATCH] Use CMAKE_INSTALL_FULL_* for installing

While I don't think there should be a difference between the two
variables for most cases, for SYSCONFDIR we might end up installing into
/usr/etc/ if we don't use the FULL_SYSCONFDIR variable.
---
 data/CMakeLists.txt                                 | 4 ++--
 doc/CMakeLists.txt                                  | 2 +-
 po/CMakeLists.txt                                   | 2 +-
 src/agent/CMakeLists.txt                            | 2 +-
 src/connectivity-api/connectivity-qt/CMakeLists.txt | 4 ++--
 src/indicator/CMakeLists.txt                        | 2 +-
 src/indicator/agent/CMakeLists.txt                  | 2 +-
 7 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 34ce2bce..27f31641 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -5,7 +5,7 @@
 
 set(
   INDICATOR_DIR
-  "${CMAKE_INSTALL_DATADIR}/unity/indicators"
+  "${CMAKE_INSTALL_FULL_DATADIR}/unity/indicators"
   CACHE FILEPATH "Indicator directory"
 )
 
@@ -73,7 +73,7 @@ add_schema ("com.lomiri.indicator.network.gschema.xml")
 ###########################
 
 # where to install
-set (XDG_AUTOSTART_DIR "${CMAKE_INSTALL_SYSCONFDIR}/xdg/autostart")
+set (XDG_AUTOSTART_DIR "${CMAKE_INSTALL_FULL_SYSCONFDIR}/xdg/autostart")
 message (STATUS "${XDG_AUTOSTART_DIR} is the DBus Service File install dir")
 
 set (XDG_AUTOSTART_NAME "${CMAKE_PROJECT_NAME}.desktop")
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 54b5c4e9..54eb05b2 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -1,4 +1,4 @@
-set(INSTALL_DOCDIR ${CMAKE_INSTALL_DOCDIR})
+set(INSTALL_DOCDIR ${CMAKE_INSTALL_FULL_DOCDIR})
 
 option(BUILD_DOC "Build documentation by default" TRUE)
 option(PRIVATE_DOCS "Build private documentation" FALSE)
diff --git a/po/CMakeLists.txt b/po/CMakeLists.txt
index dfb7d213..9decad7c 100644
--- a/po/CMakeLists.txt
+++ b/po/CMakeLists.txt
@@ -36,5 +36,5 @@ foreach(LANG ${LANGS})
                    COPYONLY)
 endforeach(LANG)
 gettext_process_pot_file(${POT_FILE} ALL
-                         INSTALL_DESTINATION ${CMAKE_INSTALL_LOCALEDIR}
+                         INSTALL_DESTINATION ${CMAKE_INSTALL_FULL_LOCALEDIR}
                          LANGUAGES ${LANGS})
diff --git a/src/agent/CMakeLists.txt b/src/agent/CMakeLists.txt
index e71aac5c..24653240 100644
--- a/src/agent/CMakeLists.txt
+++ b/src/agent/CMakeLists.txt
@@ -57,5 +57,5 @@ target_link_libraries(
 install(
   TARGETS
     lomiri-indicator-network-secret-agent
-  RUNTIME DESTINATION "${CMAKE_INSTALL_LIBEXECDIR}/lomiri-indicator-network/"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_FULL_LIBEXECDIR}/lomiri-indicator-network/"
 )
diff --git a/src/connectivity-api/connectivity-qt/CMakeLists.txt b/src/connectivity-api/connectivity-qt/CMakeLists.txt
index 663f644b..ba37ae26 100644
--- a/src/connectivity-api/connectivity-qt/CMakeLists.txt
+++ b/src/connectivity-api/connectivity-qt/CMakeLists.txt
@@ -123,7 +123,7 @@ set_target_properties(
 
 install(
     TARGETS ${CONNECTIVITY_QT_LIB_TARGET}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
 )
 
 set(PC_FILE_TARGET "${CMAKE_BINARY_DIR}/${CONNECTIVITY_QT_LIB_TARGET}.pc")
@@ -133,5 +133,5 @@ set(ABSOLUTE_SO_FILE "${CMAKE_INSTALL_FULL_LIBDIR}/lib${CONNECTIVITY_QT_LIB_TARG
 configure_file("lomiri-connectivity-qt.pc.in" ${PC_FILE_TARGET} @ONLY)
 install(
   FILES ${PC_FILE_TARGET}
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
+  DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig"
 )
diff --git a/src/indicator/CMakeLists.txt b/src/indicator/CMakeLists.txt
index ff3e0bfe..a0e3744a 100644
--- a/src/indicator/CMakeLists.txt
+++ b/src/indicator/CMakeLists.txt
@@ -174,5 +174,5 @@ target_link_libraries(
 install(
   TARGETS
     lomiri-indicator-network-service
-  RUNTIME DESTINATION "${CMAKE_INSTALL_LIBEXECDIR}/lomiri-indicator-network/"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_FULL_LIBEXECDIR}/lomiri-indicator-network/"
 )
diff --git a/src/indicator/agent/CMakeLists.txt b/src/indicator/agent/CMakeLists.txt
index 9b47bc77..38e394d8 100644
--- a/src/indicator/agent/CMakeLists.txt
+++ b/src/indicator/agent/CMakeLists.txt
@@ -73,5 +73,5 @@ target_link_libraries(
 
 install(
 	TARGETS indicator-secret-agent-bin
-	RUNTIME DESTINATION "${CMAKE_INSTALL_LIBEXECDIR}"
+	RUNTIME DESTINATION "${CMAKE_INSTALL_FULL_LIBEXECDIR}"
 )
-- 
2.43.0

